.github/workflows/install-and-test.yml

```
@@ -0,0 +1,112 @@
+# Configuration to test both CPU and GPU installations with both PDM and pip
+#
+# We don't use lock files in neural-lam, because that would tie us to a
+# particular variant of pytorch (gpu/cpu), and want it to be possible for
+# people choose which variant to install. Because we don't use lock files the
+# specific package versions to install is resolved on every install. This means
+# that has upstream packages evolve the specific version installed might change
+# (if we don't pin it). Currently, we do not pin pytorch in pyproject.toml
+#
+# To test both CPU and GPU installations with both PDM and pip we use two sets
+# of hosts to run the training on:
+# - ubuntu-latest for CPU: these are on github
+# - cirun-aws-runner--${{ github.run_id }} for GPU: these are on AWS,
+#   orchestrated by cirun.io
+#
+# To enable installing different pytorch variants we use an approach where we
+# first install a specific torch version (and variant), and then install neural-lam.
+# To ensure that the version of torch installed is compatible with neural-lam we
+# first use `pip install --dry-run .` to determine the version of torch that would
+# be installed.
+name: Test Matrix
+
+on: [push, pull_request]
+
+jobs:
+  tests:
+    runs-on: ${{ matrix.runner }}
+    strategy:
+      matrix:
+        package_manager: [pdm, pip]
+        device: [cpu, gpu]
+        include:
+          - device: cpu
+            runner: ubuntu-latest
+            torch_index: https://download.pytorch.org/whl/cpu
+          - device: gpu
+            runner: "cirun-aws-runner--${{ github.run_id }}"
+            torch_index: https://download.pytorch.org/whl/cu124
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
```

```suggestion
        uses: actions/checkout@v4
```
v2 is quite outdated with the latest patch being from spring 2023